rc/filetype/org.kak

```diff
@@ -0,0 +1,78 @@
+
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .*[.]org %{
+    set-option buffer filetype org
+}
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/org regions
+add-highlighter shared/org/inline default-region regions
+add-highlighter shared/org/inline/text default-region group
+
+evaluate-commands %sh{
+    languages="
+        c cabal clojure coffee cpp css cucumber d diff dockerfile fish gas go
+        haml haskell html ini java javascript json julia kak kickstart latex
+        lisp emacs-lisp scheme lua makefile org moon objc perl pug python ragel ruby rust
+        sass scala scss sh swift toml tupfile typescript yaml sql
+    "
+    for lang in ${languages}; do
+        printf "%s\n" "add-highlighter shared/org/${lang} region '#\+(?i)BEGIN_SRC(?I)\h+${lang}\b' '#\+(?i)END_SRC' regions"
+        printf "%s\n" "add-highlighter shared/org/${lang}/ default-region fill meta"
+        case ${lang} in
+            kak)        ref="kakrc"   ;;
+            emacs-lisp) ref="lisp"    ;;
+            *)          ref="${lang}" ;;
+        esac
+        printf "%s\n" "add-highlighter shared/org/${lang}/inner region \A#\+(?i)BEGIN_SRC(?I)[^\n]*\K '(?=#\+(?i)END_SRC)' ref ${ref}"
+    done
+}
+
+# Comment
+add-highlighter shared/org/comment region (^|\h)\K#[^+] $ fill comment
+
+# No-language SRC block
+add-highlighter shared/org/src_block region ^(\h*)(?i)#\+BEGIN_SRC(\h+|\n) ^(\h*)(?i)#\+END_SRC\h*$ fill meta
+
+# Example block
+add-highlighter shared/org/example_block region ^(\h*)(?i)#\+BEGIN_EXAMPLE(\h+|\n) ^(\h*)(?i)#\+END_EXAMPLE\h*$ fill mono
+
+# Small example block
+add-highlighter shared/org/inline/text/example regex ^(\h*)[:]\h+[^\n]* 0:mono
+
+# Unordered list items start with `-', `+', or `*'
+add-highlighter shared/org/inline/text/unordered-lists regex ^(?:\h*)([-+])\h+ 1:bullet
+# But `*' list must be indented with at least single space, if not it is treated as a heading
+add-highlighter shared/org/inline/text/star-list regex ^(?:\h+)([*])\h+ 1:bullet
+
+# Ordered list items start with a numeral followed by either a period or a right parenthesis, such as `1.' or `1)'
+add-highlighter shared/org/inline/text/ordered-lists regex ^(?:\h*)(\d+[.)])\h+ 1:bullet
+
+# Headings
+add-highlighter shared/org/inline/text/heading    regex "^[*]+\h+[^\n]+"                          0:header
+add-highlighter shared/org/inline/text/italic     regex "\s([/][^\s/].*?[^\s/]*?[/])\W" 1:default,default+i
+add-highlighter shared/org/inline/text/verbatim   regex "\s([=][^\s=].*?[^\s=]*?[=])\W" 1:meta
+add-highlighter shared/org/inline/text/code       regex "\s([~][^\s~].*?[^\s~]*?[~])\W" 1:mono
+add-highlighter shared/org/inline/text/math       regex "\s([$][^\s].*?[^\s]*?[$])\W"   1:mono
+add-highlighter shared/org/inline/text/underlined regex "\s([_][^\s_].*?[^\s_]*?[_])\W" 1:default,default+u
+add-highlighter shared/org/inline/text/bold       regex "\s([*][^\s*].*?[^\s*]*?[*])\W" 1:default,default+b
```

Comment:
There are existing, standard faces named `italic` and `bold` you can use instead of `default,default+i` and `default,default+b`. I'm surprised there isn't already one for `underlined`, but it shouldn't be hard to add.

Reply:
I didn't use those faces because org mode in Emacs doesn't colors those, but adds attribute to the text. This way nesting is also possible, like bold links that still use link face.

Reply:
I get the impression that many/most Emacs users run it in a GUI mode so they can get all the fancy formatting features, so they can assume that those attributes are available without any kind of fallback. Since Kakoune is primarily a terminal-based editor, it can't safely assume those attributes will always be available, so it's good to have the fallback of Kakoune's standard colors.

That said, nesting *should* be possible in Kakoune, since attributes aren't mutually exclusive. It doesn't seem to work in the Kakoune ncurses UI, though: if I create a Markdown document with:

```markdown
A test of **some bold _with italics_ text**
```

the text "with italics" is red and italic but not bold in regular Kakoune, but red and bold-italic in my experimental `kakoune-gtk`. How strange.

Reply:
> I get the impression that many/most Emacs users run it in a GUI mode so they can get all the fancy formatting features, so they can assume that those attributes are available without any kind of fallback.

The _markup symbols_ itself are fallback, e.g. /*italic text*/ looks nice, but if attribute isn't available one can still distinguish /italic text/ (kinda old convention for italic formatting) from non-italic. Same goes for \***bold text**\*. I mean, it's not Markdown, where we really need to track the count of `*` or `_` to find out if it is providing any attribute to the text.

Reply:
> That said, nesting should be possible in Kakoune, since attributes aren't mutually exclusive. It doesn't seem to work in the Kakoune ncurses UI, though

My main concern is that bold links and must be colored as links with bold attribute. I guess this can be specified in the theme though.

Reply:
Yeah, the markup symbols themselves are fallbacks, but I think an Org-mode plugin that ships with Kakoune should follow the same conventions as all the other plugins that ship with Kakoune. If it were a third-party plugin, it wouldn't matter so much (although if it were me, I'd still follow the Kakoune conventions rather than the Org-mode conventions).

Reply:
Got it. Well, I guess the perfect solution would be to use faces, and override those faces in user configuration then.